e2e/feature/dashboard/header/step_definitions/headerFeature_steps.ts

```diff
@@ -0,0 +1,99 @@
+import { browser, element, by } from 'protractor';
+import { HeaderPage } from '../../../../page-objects/header.po';
+import { Capdash2Page } from '../../../../page-objects/app.po';
+import { E2EHelpers } from '../../../../support/e2eHelpers';
+
+import { defineSupportCode } from 'cucumber';
+
+import { expect } from 'chai';
+
+defineSupportCode(({Given, When, Then}) => {
+  const header: HeaderPage = new HeaderPage();
+  const app: Capdash2Page = new Capdash2Page();
+
+  const menus = [
+    {
+      name: 'mega-menu',
+      select: () => { header.clickMegaMenu() },
+      element: header.dropdownHeader
+    },
+    {
+      name: 'notifications',
+      select: () => { header.clickNotificationsMenu() },
+      element: header.mediaList
+    },
+    {
+      name: 'languages',
+      select: () => { header.clickLanguagesMenu() },
+      element: header.languagesList
+    },
+    {
+      name: 'user',
+      select: () => { header.clickUserMenu() },
+      element: header.logout
+    }
+  ]
+
+  Given('the Capacity Dashboard header is loaded', () => {
+    return app.getAppHeader().then( appHeader => {
+      appHeader.isDisplayed().then(isDisplayed => {
+        expect(isDisplayed).to.equal(true);
+      });
+    });
+  });
+
+  menus.forEach( menu => {
+
+    When(new RegExp('the user selects the ' + menu.name + ' menu'), () => {
```

Comment:
Instead of using a .forEach can use the cucumber built-in that lets you leverage the Examples section:
When(n'the user selects the {menu} menu', (menu) => {